extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java

```diff
@@ -37,28 +41,50 @@
     /** RegEx to extract a scale definition */
     private static final Pattern LIMITS_PATTERN = Pattern.compile("(\\[|\\])(.*)\\.\\.(.*)(\\[|\\])");
 
+    class OrderedProperties extends Properties {
+        private static final long serialVersionUID = 3860553217028220119L;
+        private final HashSet<Object> keys = new LinkedHashSet<Object>();
+
+        public OrderedProperties() {
+        }
+
+        public Iterable<Object> orderedKeys() {
+            return Collections.list(keys());
```

Comment:
I think that we can also return `keys` direct;ly I do not yet see the added value of the current solution:

```java
Set<Object> orderedKeys() {
   return keys;
}
```